snuba/web/split.py

```diff
@@ -57,40 +59,68 @@ def time_split(dataset, request: Request, *args, **kwargs):
+        from_date_str = next(
+            (
+                condition[2]
+                for condition in conditions
+                if _identify_condition(condition, self.__timestamp_col, ">=")
+            ),
+            None,
+        )
+
+        to_date_str = next(
+            (
+                condition[2]
+                for condition in conditions
+                if _identify_condition(condition, self.__timestamp_col, "<")
+            ),
+            None,
+        )
+
+        if not from_date_str or not to_date_str:
+            return None
+
         date_align, split_step = state.get_configs(
             [("date_align_seconds", 1), ("split_step", 3600)]  # default 1 hour
         )
+        to_date = util.parse_datetime(to_date_str, date_align)
+        from_date = util.parse_datetime(from_date_str, date_align)
 
-        query_limit = request.query.get_limit()
-        limit = query_limit if query_limit is not None else 0
         remaining_offset = request.query.get_offset()
 
-        to_date = util.parse_datetime(
-            request.extensions["timeseries"]["to_date"], date_align
-        )
-        from_date = util.parse_datetime(
-            request.extensions["timeseries"]["from_date"], date_align
-        )
-
         overall_result = None
         split_end = to_date
         split_start = max(split_end - timedelta(seconds=split_step), from_date)
         total_results = 0
         while split_start < split_end and total_results < limit:
-            request.extensions["timeseries"]["from_date"] = split_start.isoformat()
-            request.extensions["timeseries"]["to_date"] = split_end.isoformat()
+            # We need to make a copy to use during the query execution because we replace
+            # the start-end conditions on the query at each iteration of this loop.
+            split_request = copy.deepcopy(request)
+
+            _replace_condition(
+                split_request.query, self.__timestamp_col, ">=", split_start.isoformat()
+            )
+            _replace_condition(
+                split_request.query, self.__timestamp_col, "<", split_end.isoformat()
+            )
             # Because its paged, we have to ask for (limit+offset) results
             # and set offset=0 so we can then trim them ourselves.
-            request.query.set_offset(0)
-            request.query.set_limit(limit - total_results + remaining_offset)
+            split_request.query.set_offset(0)
+            split_request.query.set_limit(limit - total_results + remaining_offset)
 
-            # The query function may mutate the request body during query
-            # evaluation, so we need to copy the body to ensure that the query
-            # has not been modified in between this call and the next loop
-            # iteration, if needed.
-            # XXX: The extra data is carried across from the initial response
-            # and never updated.
```

Comment:
This is still true, nothing has changed. If we're going to remove this reminder it should go on the debt list.

This comment wasn't fixed, but marked as resolved. https://github.com/getsentry/snuba/pull/865#discussion_r404553324

Reply:
To be fair I removed the comment because I found it more confusing than helpful, (thus also marked the comment as resolved). I can bring it back if you find useful.

Reply:
I'll put it on the debt list but I don't understand that thought process to be honest.

Reply:
Will rewrite it with more details to be more explicit.

Reply:
added the comment back with more details.